src/cmap/auth/mongodb_oidc.ts

```diff
@@ -40,14 +44,9 @@ export type OIDCRefreshFunction = (
 ) => Promise<OIDCRequestTokenResult>;
 
 /** @internal */
-export const OIDC_WORKFLOWS = {
-  callback: new CallbackWorkflow(),
-  aws: new AwsDeviceWorkflow(),
-  azure: undefined,
-  gcp: undefined,
-  // Set the protype to null to avoid false returns and keep types.
-  __proto__: null
-};
+export const OIDC_WORKFLOWS: Map<string, Workflow> = new Map();
```

Comment:
should we provide a better type here and restrict the allowed keys in this map?

Reply:
Needs to be a generic string here since that's what `SERVICE_NAME` is. If we do change it again to what Anna suggested in https://github.com/mongodb/specifications/pull/1383 then I'll make this more strict again in the NODE-5036 PR not to hold this up.

Reply:
I think it would be easier to just make this change here, since `PROVIDER_NAME` seems to be the consensus - I can merge it first thing tomorrow morning once it's all updated, since the semantics of the rest of this PR shouldn't be affected

Reply:
Change is made here now.